pkg/network/payload/headers.go

```diff
@@ -22,7 +21,6 @@ func (p *Headers) DecodeBinary(br *io.BinReader) {
 
 	// C# node does it silently
 	if lenHeaders > MaxHeadersAllowed {
-		log.Warnf("received %d headers, capping to %d", lenHeaders, MaxHeadersAllowed)
```

Comment:
Not sure it's a good idea to drop it. Downplaying it from warning may be useful, but dropping completely?..

Reply:
I think it is better when encode/decode code is free of side effects.

I have come up with 2 other solutions:
1. Make a field `bool Headers.capped` and do some logging in caller.
2. Set an error in `BinaryReader`, then log and nullify it if it is `ErrMaxHeadersExceeded`

Reply:
Technically it'd be correct to set a `br.Err` because we've not decoded the payload completely here, but I'm not sure how it would complicate the error path above this decoder.

Reply:
I set `Received` field on decoding and later check if it is consistent with provided array. This seems the cleanest solution to me, as it doesn't pollute main code path with the details of the specific command.